subprojects/gst-plugins-good/gst/rtpmanager/rtptwccstats.c

```diff
@@ -367,9 +369,17 @@ _keep_history_length (TWCCStatsManager *statsman, gsize max_len,
           "sent_packets FIFO overflows, dropping");
     } else if (GST_CLOCK_TIME_IS_VALID (statsman->prev_stat_window_beginning) &&
         GST_CLOCK_DIFF (pkt_ts, statsman->prev_stat_window_beginning)
-        < GST_MSECOND * 1500) {
-      GST_WARNING_OBJECT (statsman->parent, "Risk of"
-          " underrun of sent_packets FIFO");
+        < GST_MSECOND * 1500 &&
+        GST_CLOCK_TIME_IS_VALID (cur_time)) {
+          const gboolean enough_time_since_last_record = 
+            !GST_CLOCK_TIME_IS_VALID (statsman->underrun_tx_pkt_log_ts)
+            || GST_CLOCK_DIFF (statsman->underrun_tx_pkt_log_ts, cur_time) 
+              > GST_SECOND * 2;
+          if (enough_time_since_last_record) {
+            statsman->underrun_tx_pkt_log_ts = cur_time;
+            GST_WARNING_OBJECT (statsman->parent, "Risk of"
+              " underrun of sent_packets FIFO");
```

Comment:
given we don't always log it. would it be useful to include the sent_packets queue length here?